src/theory/fp/theory_fp_rewriter.cpp

```diff
@@ -69,7 +68,17 @@ namespace rewrite {
   RewriteResponse removeDoubleNegation (TNode node, bool) {
     Assert(node.getKind() == kind::FLOATINGPOINT_NEG);
     if (node[0].getKind() == kind::FLOATINGPOINT_NEG) {
-      RewriteResponse(REWRITE_AGAIN, node[0][0]);
+      return RewriteResponse(REWRITE_AGAIN, node[0][0]);
+    }
+
+    return RewriteResponse(REWRITE_DONE, node);
+  }
+
+  RewriteResponse compactAbs (TNode node, bool) {
+    Assert(node.getKind() == kind::FLOATINGPOINT_ABS);
+    if (node[0].getKind() == kind::FLOATINGPOINT_NEG ||
+	node[0].getKind() == kind::FLOATINGPOINT_ABS) {
+      return RewriteResponse(REWRITE_AGAIN, node[0][0]);
```

Comment:
We could apply a similar rewrite in other cases such as removing all signs in a multiplication. E.g. `abs(-a * b) --> abs(a * b)`. This might be for another PR though.

Reply:
Yeah... there are a lot of sign related rewrite but I think I want a better way of approaching them so it will be a different PR.